mechanicalsoup/browser.py

```diff
@@ -181,7 +181,11 @@ def _request(self, form, url=None, **kwargs):
                     # (or submit button) enctype attribute is set to
                     # "multipart/form-data". We don't care, simplify.
                     if not value:
-                        continue
+                        # create a temporary file for upload
+                        file_path = tempfile.mkstemp()[1]
+                        with open(file_path, "w") as f:
+                            f.write("")
```

Comment:
Creating a temporary file is a very heavyweight way to submit an empty value. In the end, we don't send a "file" but it's content. It should be possible to submit an empty file without actually creating it locally.

Reply:
Indeed, I've tested this, and so long as we are sending the empty string (e.g. `files[name] = ""`), then it will be part of the submitted request. `None` doesn't work as a value, but I think we don't have to worry about that since we use `value = tag.get("value", "")`, which should always be a string.